ldi-core/ldes-client/tree-node-supplier/src/main/java/ldes/client/treenodesupplier/repository/filebased/FileBasedMemberRepository.java

```diff
@@ -0,0 +1,57 @@
+package ldes.client.treenodesupplier.repository.filebased;
+
+import ldes.client.treenodesupplier.domain.entities.MemberRecord;
+import ldes.client.treenodesupplier.domain.valueobject.MemberStatus;
+import ldes.client.treenodesupplier.repository.MemberRepository;
+import ldes.client.treenodesupplier.repository.filebased.mapper.MemberRecordMapper;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class FileBasedMemberRepository implements MemberRepository {
+	public static final String UNPROCESSED_MEMBERS = "membersUnprocessed.txt";
+	public static final String PROCESSED_MEMBERS = "membersProcessed.txt";
+
+	private final FileManagerFactory fileManagerFactory = FileManagerFactory.getInstance();
+	private final FileManager fileManager = fileManagerFactory.getFileManager();
+	private final MemberRecordMapper mapper = new MemberRecordMapper();
+
+	@Override
+	public Optional<MemberRecord> getUnprocessedTreeMember() {
+		List<MemberRecord> unprocessed = fileManager.getRecords(UNPROCESSED_MEMBERS).map(mapper::toMemberRecord)
```

Comment:
How big can this file get? If this is a big file, I feel reading everything when you only need the first line is quite inefficient? Maybe a fileManger.getFirst() method could be useful?

Reply:
Max size would be one tree node (since once the members are processed, they are deleted from unprocessed and put into processed).